ui_framework/components/color_picker/color_picker.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { ChromePicker } from 'react-color';
+
+export class KuiColorPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showColorSelector: false,
+    };
+  }
+
+  closeColorSelector = () => {
+    this.setState({ showColorSelector: false });
+  };
+
+  toggleColorSelector = () => {
+    this.setState({ showColorSelector: !this.state.showColorSelector });
+  };
+
+  handleColorSelection = (color) => {
+    this.props.onChange(color.hex);
+  };
+
+  render() {
```

Comment:
I spoke with @zinckiwi and he suggested a way to support closing the colorpicker without adding an element which will intercept the mouse click.  Here's the code:

```jsx
onClickRootElement = e => {
    // This prevents clicking on the element from closing it, due to the event handler on the
    // document object.
    e.nativeEvent.stopImmediatePropagation();
  }

  componentDidMount() {
    // When the user clicks somewhere outside of the color picker, we will dismiss it.
    document.addEventListener('click', this.closeColorSelector);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.closeColorSelector);
  }

  render() {
    const color = this.props.color || '#ffffff';
    const classes = classNames('kuiColorPicker', this.props.className);
    return (
      <div
        className={ classes }
        aria-label={ this.props['aria-label'] }
        data-test-subj={ this.props['data-test-subj'] }
        onClick={ this.onClickRootElement }
      >
        <div
          className="kuiColorPicker__preview"
          onClick={ this.toggleColorSelector }
        >
          <div
            className="kuiColorPicker__swatch"
            aria-label="Select a color"
            data-test-subj="colorSwatch"
            style={{ background: this.props.color }}
          />
          <div
            className="kuiColorPicker__label"
            aria-label={`Color selection is ${color}`}
          >
            { color }
          </div>
        </div>
        {
          this.state.showColorSelector ?
            <div className="kuiColorPickerPopUp" data-test-subj="colorPickerPopup">
              <ChromePicker
                color={ color }
                disableAlpha={ true }
                onChange={ this.handleColorSelection }
              />
            </div>
            : null
        }
      </div>
    );
```

With this change we add a click handler to the document which will close the color picker. When we click on the color picker itself, we prevent the event from reaching this handler; otherwise clicking the color picker would open and then immediately close it. What do you think @Stacey-Gammon?

@zinckiwi Can you confirm this is what you had in mind?